applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_strain_energy_response_function.h

```diff
@@ -0,0 +1,215 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder 
+//   
+
+#ifndef ADJOINT_STRAIN_ENERGY_RESPONSE_FUNCTION_H
+#define ADJOINT_STRAIN_ENERGY_RESPONSE_FUNCTION_H
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// External includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "adjoint_structural_response_function.h"
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) AdjointStrainEnergyResponseFunction : public AdjointStructuralResponseFunction
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    typedef AdjointStructuralResponseFunction BaseType;
+
+    /// Pointer definition of AdjointStrainEnergyResponseFunction
+    KRATOS_CLASS_POINTER_DEFINITION(AdjointStrainEnergyResponseFunction);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    AdjointStrainEnergyResponseFunction(ModelPart& rModelPart, Parameters ResponseSettings);
+
+    /// Destructor.
+    ~AdjointStrainEnergyResponseFunction();
+   
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Initialize() override;
+
+    double CalculateValue(ModelPart& rModelPart) override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const;
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream &rOStream) const;
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream &rOStream) const;
+
```

Comment:
@armingeiser are these three functions still used somewhere or can we delete them?

Reply:
Done